File: 3111-minimum-rectangles-to-cover-points/3111-minimum-rectangles-to-cover-points.java
class Solution {
  public int minRectanglesToCoverPoints(int[][] points, int w) {
    Arrays.sort(points, (a,b)-> a[0]-b[0]);
    int count = 1;

    int threshold = points[0][0] + w;
    for (int[] point : points) {
      if (point[0] > threshold) {
        count++;
        threshold = point[0] + w;
      }
    }

    return count;
  }
}

File: 3111-minimum-rectangles-to-cover-points/3111-minimum-rectangles-to-cover-points.cpp
class Solution {
public:
    int minRectanglesToCoverPoints(vector<vector<int>>& points, int w) {
        int n=points.size();
        vector<int> temp(n);
        for(int i=0;i<n;i++)
        {
            temp[i]=points[i][0];
        }
        sort(temp.begin(),temp.end());
        int j=0,count=0;
        for(int i=0;i<n;i++)
        {
            if(temp[i]-temp[j]<=w)
                continue;
            else
            {
                count++;
                j=i;
            }
        }
        count++;
        return count;
    }
};